conseil-api/src/main/scala/tech/cryptonomic/conseil/api/routes/platform/data/ethereum/EthereumDataRoutesCreator.scala

```diff
@@ -99,14 +99,77 @@ trait EthereumDataRoutesCreator
       }
   }
 
+  /** V2 Route implementation for logs endpoint */
+  private val logsRoute: Route = ethereumLogsEndpoint.implementedByAsync {
+    case ((network, filter), _) =>
+      platformNetworkValidation(network) {
+        operations.fetchLogs(filter.toQuery.withLimitCap(maxQueryResultSize))
+      }
+  }
+
+  /** V2 Route implementation for receipts endpoint */
+  private val receiptsRoute: Route = ethereumReceiptsEndpoint.implementedByAsync {
+    case ((network, filter), _) =>
+      platformNetworkValidation(network) {
+        operations.fetchReceipts(filter.toQuery.withLimitCap(maxQueryResultSize))
+      }
+  }
+
+  /** V2 Route implementation for contracts endpoint */
+  private val contractsRoute: Route = ethereumContractsEndpoint.implementedByAsync {
+    case ((network, filter), _) =>
+      platformNetworkValidation(network) {
+        operations.fetchContracts(filter.toQuery.withLimitCap(maxQueryResultSize))
+      }
+  }
+
+  /** V2 Route implementation for tokens endpoint */
+  private val tokensRoute: Route = ethereumTokensEndpoint.implementedByAsync {
+    case ((network, filter), _) =>
+      platformNetworkValidation(network) {
+        operations.fetchTokens(filter.toQuery.withLimitCap(maxQueryResultSize))
+      }
+  }
+
+  /** V2 Route implementation for token transfers endpoint */
+  private val tokenTransfersRoute: Route = ethereumTokenTransfersEndpoint.implementedByAsync {
+    case ((network, filter), _) =>
+      platformNetworkValidation(network) {
+        operations.fetchTokenTransfers(filter.toQuery.withLimitCap(maxQueryResultSize))
+      }
+  }
+
+  /** V2 Route implementation for accounts endpoint */
+  private val accountsRoute: Route = ethereumAccountsEndpoint.implementedByAsync {
+    case ((network, filter), _) =>
+      platformNetworkValidation(network) {
+        operations.fetchAccounts(filter.toQuery.withLimitCap(maxQueryResultSize))
+      }
+  }
+
+  /** V2 Route implementation for accounts by address endpoint */
+  private val accountsByAddressRoute: Route = ethereumAccountByAddressEndpoint.implementedByAsync {
+    case ((network, address), _) =>
+      platformNetworkValidation(network) {
+        operations.fetchAccountByAddress(address)
+      }
+  }
+
```

Comment:
More of a philosophical question than anything else... I look at this code and think...
what actually changes between the calls...? Why do we copy paste it? Is it clearer? Is it better?

Or is a single abstraction called `delegateToOperations(operationCall: in => out)` clearer and cleaner?

Reply:
Well... It is not so easy, mostly because different endpoints takes (or produces) different arguments. Sometimes you have an address, sometimes you have a filter, but maybe that can be extracted (I'm not sure if that is possible with this weird types in Endpoints4s - to find generic enough). 

Additionally, I see only one think that could be extracted, which is `platformNetworkValidation`. If you look at the documentation from Endpoints4s ([Authentication part](https://endpoints4s.github.io/guides/custom-authentication.html) to be specific), you will see there is a generic way of checking API-Key. I believe there is possibility to do the same thing but with `platformNetworkValidation` - to extract this from this place and implement it only once, cause `network` is all you need and you will have it always.

This being done properly, will reduce number of parameters here (apiKey that we have here which is mapped to `_` could be also removed). We could also get rid of this `platformNetworkValidation` and the only think left would be
`ethereumAccountByAddressEndpoint.implementedByAsync(address => operations.fetchAccountByAddress(address))` which is an essence of this route. You cannot reduce it more :) 

So in short we could try do, what you want, but that would be middle step. The change I'm talking about is a bit more complex tho, and can take couple of days to implement.

Reply:
I think it might be worth it, but the end call's yours

Reply:
It might, be we are talking about re-designing 40% of the current API's implementation :D

Reply:
😮 ... I guess we have a clear choice then... 😬